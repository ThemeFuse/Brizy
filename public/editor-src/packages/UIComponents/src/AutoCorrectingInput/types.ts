import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
