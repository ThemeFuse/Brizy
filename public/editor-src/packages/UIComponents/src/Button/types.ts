import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
