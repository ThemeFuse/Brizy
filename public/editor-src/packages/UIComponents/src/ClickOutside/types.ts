import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
