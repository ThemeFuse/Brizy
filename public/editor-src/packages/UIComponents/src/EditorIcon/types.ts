export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
