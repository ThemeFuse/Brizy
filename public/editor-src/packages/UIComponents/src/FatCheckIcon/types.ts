import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
