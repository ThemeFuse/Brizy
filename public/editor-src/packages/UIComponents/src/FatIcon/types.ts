import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
