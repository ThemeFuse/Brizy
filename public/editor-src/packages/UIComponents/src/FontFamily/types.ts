import { FontFamilyType } from "../types/fonts/familyType";
import { Weight } from "../types/fonts/weight";

export type FontsBlock = Partial<{
  config: Array<FontObject>;
  blocks: Array<FontObject>;
  [FontFamilyType.google]: Array<FontObject>;
  [FontFamilyType.upload]: Array<FontObject>;
}>;

export interface Props {
  addFont?: VoidFunction;
  onChange: (font: FontWithType) => void;
  value: string;
  className?: string;
  addFontLabel?: string;
  fonts: FontsBlock;
}
export interface FontSizes {
  default: string;
  great_vibes: string;
  alex_brush: string;
  allura: string;
  parisienne: string;
}

export interface FontObject {
  brizyId?: string;
  family: string;
  id: string;
  title: string;
  weights: Array<Weight>;
}

export interface FontWithType {
  id: string;
  type: FontFamilyType;
  weights: Array<Weight>;
}

export interface FontList {
  fonts: Array<FontObject>;
  type: FontFamilyType;
}
