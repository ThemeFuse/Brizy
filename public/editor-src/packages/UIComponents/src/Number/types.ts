import { WithClassName, WithSize } from "../types/attributes";

export type Action = "increase" | "decrease" | "none";

export interface Props extends WithClassName, WithSize {
  value: number | undefined;
  onChange: (v: number | undefined) => void;
  onIncrease: VoidFunction;
  onDecrease: VoidFunction;
  showArrows?: boolean;
}
