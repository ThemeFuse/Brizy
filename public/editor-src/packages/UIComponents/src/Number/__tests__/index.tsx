import { fireEvent, render } from "@testing-library/react";
import React from "react";
import _ from "underscore";
import { Size } from "../../types/attributes";
import { Number } from "../index";

describe("Test Number", () => {
  describe("Snapshots", () => {
    test("By default showArrows should be true and size auto", () => {
      const { container } = render(
        <Number
          onChange={_.noop}
          onDecrease={_.noop}
          onIncrease={_.noop}
          value={undefined}
        />
      );

      const numberDiv = container.querySelector(".brz-ed-control__number");
      const arrows = numberDiv?.querySelector(
        ".brz-ed-control__number--arrows"
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(numberDiv).toHaveClass("brz-ed-control__number--auto");
      expect(arrows).toBeInTheDocument();
    });
    test.each([0, 1, -1])("value should be %s", (value) => {
      const { container } = render(
        <Number
          onChange={_.noop}
          onDecrease={_.noop}
          onIncrease={_.noop}
          value={value}
        />
      );

      const input = container.querySelector(".brz-input");

      expect(container.firstChild).toMatchSnapshot();

      expect(input?.getAttribute("value")).toBe(value.toString());
    });
    test.each<Size>(["auto", "short", "medium", "large"])(
      "size should be %s",
      (size) => {
        const { container } = render(
          <Number
            onChange={_.noop}
            onDecrease={_.noop}
            onIncrease={_.noop}
            size={size}
            value={undefined}
          />
        );

        const numberDiv = container.querySelector(".brz-ed-control__number");

        expect(container.firstChild).toMatchSnapshot();

        expect(numberDiv).toHaveClass(`brz-ed-control__number--${size}`);
      }
    );
    test("className should be className", () => {
      const { container } = render(
        <Number
          className="className"
          onChange={_.noop}
          onDecrease={_.noop}
          onIncrease={_.noop}
          value={undefined}
        />
      );
      const numberDiv = container.querySelector(".brz-ed-control__number");

      expect(container.firstChild).toMatchSnapshot();

      expect(numberDiv).toHaveClass("className");
    });
  });

  describe("Check event handlers", () => {
    test("Check change event", () => {
      const onChange = jest.fn();
      const { container } = render(
        <Number
          onChange={onChange}
          onDecrease={_.noop}
          onIncrease={_.noop}
          showArrows
          value={0}
        />
      );
      const input = container.querySelector(".brz-input");
      input && fireEvent.change(input, { target: { value: "1" } });

      expect(onChange).toHaveBeenCalledWith(1);
    });
    test("Check decrease event", () => {
      const onDecrease = jest.fn();
      const { container } = render(
        <Number
          onChange={_.noop}
          onDecrease={onDecrease}
          onIncrease={_.noop}
          showArrows
          value={0}
        />
      );
      const ArrowDown = container?.querySelector(
        ".brz-ed-control__number--down"
      );
      ArrowDown && fireEvent.click(ArrowDown);

      expect(onDecrease).toHaveBeenCalledTimes(1);
    });
    test("Check increase event", () => {
      const onIncrease = jest.fn();
      const { container } = render(
        <Number
          onChange={_.noop}
          onDecrease={_.noop}
          onIncrease={onIncrease}
          showArrows
          value={0}
        />
      );
      const ArrowUp = container?.querySelector(".brz-ed-control__number--up");
      ArrowUp && fireEvent.click(ArrowUp);

      expect(onIncrease).toHaveBeenCalledTimes(1);
    });
  });
});
