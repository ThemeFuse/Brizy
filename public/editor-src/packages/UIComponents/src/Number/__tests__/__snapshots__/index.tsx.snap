// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Number Snapshots By default showArrows should be true and size auto 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--auto"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value=""
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots className should be className 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--auto className"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value=""
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots size should be auto 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--auto"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value=""
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots size should be large 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--large"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value=""
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots size should be medium 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--medium"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value=""
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots size should be short 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--short"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value=""
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots value should be -1 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--auto"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value="-1"
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots value should be 0 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--auto"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value="0"
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;

exports[`Test Number Snapshots value should be 1 1`] = `
<div
  class="brz-ed-control__number brz-ed-control__number--auto"
>
  <input
    class="brz-input"
    max="999999"
    min="-999999"
    type="number"
    value="1"
  />
  <div
    class="brz-ed-control__number--arrows"
  >
    <div
      class="brz-ed-control__number--up"
    />
    <div
      class="brz-ed-control__number--down"
    />
  </div>
</div>
`;
